pkg/apis/cassandraoperator/v1alpha1/cassandradatacenter.go

```diff
@@ -10,18 +10,17 @@ import (
 type CassandraDataCenterSpec struct {
 	// Cluster is either a string or v1.LocalObjectReference
 	//Cluster interface{} `json:"cluster,omitempty"`
-	Cluster          string                    `json:"cluster,omitempty"`
-	Nodes            int32                     `json:"nodes"`
-	CassandraImage   string                    `json:"cassandraImage"`
-	SidecarImage     string                    `json:"sidecarImage"`
-	ImagePullPolicy  v1.PullPolicy             `json:"imagePullPolicy"`
-	ImagePullSecrets []v1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
-
-	Resources v1.ResourceRequirements `json:"resources"`
-
+	Cluster             string                       `json:"cluster,omitempty"`
+	Nodes               int32                        `json:"nodes"`
+	CassandraImage      string                       `json:"cassandraImage"`
+	SidecarImage        string                       `json:"sidecarImage"`
+	ImagePullPolicy     v1.PullPolicy                `json:"imagePullPolicy"`
+	ImagePullSecrets    []v1.LocalObjectReference    `json:"imagePullSecrets,omitempty"`
+	BackupSecrets       string                       `json:"backupSecrets,omitempty"`
```

Comment:
`ImagePullSecrets` are `v1.LocalObjectReference`s. Should this be the same?

Reply:
yes, changing

Reply:
redesigned to use secret volumes